.github/pull_request_template.md

```diff
@@ -1,13 +1,24 @@
 # Objective
 
-- Describe the objective or issue this PR addresses.
-- If you're fixing a specific issue, say "Fixes #X".
+Describe the objective or issue this PR addresses.
 
-## Solution
+If you're fixing a specific issue, say "Fixes #X" and the linked issue will automatically be closed when this PR is merged.
+Under each bullet point, describe how this change addressed those objectives if it is not obvious.
 
-- Describe the solution used to achieve the objective above.
+**Changes that will affect external library users must update RELEASES.md before they will be merged.**
 
----
+## Context
+
+Discuss any context that may be needed for a user with only passing acquaintance with this library to understand the changes you've made.
```

Comment:
Nit: I think "passing acquaintance" is not as clear and only applies to newcomers (as opposed to people who are experienced in other parts of the library but maybe not this particular area).
```suggestion
Discuss any context that may be needed for a user who is not as familiar with this library to understand the changes you've made.
```